plugins/collections/get.js

```diff
@@ -7,42 +7,103 @@ const getSchema = schema.models.collection.get;
 const idSchema = joi.reach(schema.models.collection.base, 'id');
 
 /**
- * Helper function to get last builds of pipelines
+ * Helper function to get PR info of pipeline
  *
- * @param   {Array}    pipelines     Array of pipeline models to get last builds for
- * @param   {Factory}  eventFactory  Factory for getting last event model
- * @returns {Array}
+ * @param   {Pipeline} pipeline  Pipeline model to get PR info for
+ * @returns {Object}
  */
-function getPipelinesHealth(pipelines, eventFactory) {
-    return pipelines
-        // Filter out pipelines that don't exist
-        .filter(pipeline => !!pipeline)
-        .map((pipeline) => {
-            const result = Object.assign({}, pipeline.toJson());
+function getPipelinePRInfo(pipeline) {
+    const prs = {
+        open: 0,
+        failing: 0
+    };
 
-            if (!pipeline.lastEventId) {
-                return result;
+    // Get all the PR jobs
+    return pipeline.getJobs({ type: 'pr' })
+        .then(prJobs => (
+            // Return array of prJobs' builds
+            Promise.all(prJobs.map(job =>
+                job.getBuilds()
+            ))
+        ))
+        .then((prJobsBuilds) => {
+            // prJobsBuilds is a 2D array where each element is an array of builds
+            // for one PR job.
+            prs.open = prJobsBuilds.length;
+
+            prJobsBuilds.forEach((jobBuilds) => {
+                // Check the first element (last build) if pr job failed
+                if (jobBuilds[0].status === 'FAILURE') {
+                    prs.failing += 1;
+                }
+            });
+
+            return prs;
+        })
+        .catch(() => prs);
```

Comment:
Just curious, if we do the error handling like this, can we see the error get caught in the server log? Otherwise it can be hard for debugging

Reply:
Refactored it to use `winston` to log any unexpected errors on the server so we can at least debug them on the server side if they happen.